internal/tpm/tpm.go

```diff
@@ -1,20 +1,60 @@
 package tpm
 
 import (
+	"bytes"
 	"encoding/hex"
 	"fmt"
 	"io"
+	"os"
 
+	"github.com/google/go-tpm-tools/client"
+	pbattest "github.com/google/go-tpm-tools/proto/attest"
+	pbtpm "github.com/google/go-tpm-tools/proto/tpm"
 	"github.com/google/go-tpm-tools/simulator"
 	"github.com/google/go-tpm/legacy/tpm2"
 	"github.com/google/go-tpm/tpmutil"
 )
 
+const MinNonceLength = 8
+const TpmSystemPath = "/dev/tpm/tpm0"
+const TpmVersionInfoPath = "/sys/class/tpm/tpm0/tpm_version_major"
+
 type TPM struct {
 	devicePath string
 	channel    io.ReadWriteCloser
 }
 
+// Note: this may be a hardware TPM or a software or emulated TPM available to the system
+func TpmExists() bool {
+	if _, err := os.Stat(TpmSystemPath); err == nil {
+		return true
+	}
+	return false
+}
+
+func TpmIsVersion2() error {
+	if !TpmExists() {
+		return fmt.Errorf("No TPM detected at %s", TpmSystemPath)
+	}
+	versionBytes, err := os.ReadFile(TpmVersionInfoPath)
+	if err != nil {
+		return err
+	}
+	versionStr := string(bytes.TrimSpace(versionBytes))
+	if versionStr != "2" {
+		return fmt.Errorf("TPM is not version 2.0")
```

Comment:
same

Reply:
this one is just in caps because `TPM` is always in caps